### PR TITLE
fix ngm crash when PD is unavailable

### PR DESCRIPTION
**Pull request by crazycs520**
close https://github.com/pingcap/ng-monitoring/issues/23

### CODE REVIEW

**Comment by crazycs520** [comment]
@zhongzc PTAL

**Comment by zhongzc** [comment]
Please fix conflicts

File: component/domain/client.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	pdCli, err := CreatePDClient(cfg)
+	if err != nil {
+		return nil, nil, err
+	}
+	return pdCli, etcdCli, err
+}
+
+func CreatePDClient(cfg *config.Config) (*pdclient.APIClient, error) {
```
Comment (zhongzc) [inline component/domain/client.go:142]: how about not to export this function?

File: component/domain/domain.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	}
+}
+
+func (do *Domain) createClientWithRetry() error {
+	if do.cm.IsInitialized() {
+		if !do.cm.NeedRecreateClient(config.GetGlobalConfig().PD) {
+			return nil
+		}
+		do.cm.Close()
+		do.cm = NewClientMaintainer()
```
Comment (zhongzc) [inline component/domain/domain.go:73]: May leak something?

File: component/domain/client.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		Context:   context.Background(),
+		TLS:       cfg.Security.GetTLSConfig(),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pdCli, err := CreatePDClient(cfg)
+	if err != nil {
+		return nil, nil, err
```
Comment (zhongzc) [inline component/domain/client.go:137]: need to close etcdCli

File: component/domain/domain.go (hunk 2 above)
Comment (crazycs520) [reply component/domain/domain.go:73]: why?

File: component/domain/client.go (hunk 3 above)
Comment (crazycs520) [reply component/domain/client.go:137]: nice catch

File: component/domain/client.go (hunk 1 above)
Comment (crazycs520) [reply component/domain/client.go:143]: No, this function will be used in other pkg test.

File: component/domain/domain.go (hunk 2 above)
Comment (crazycs520) [reply component/domain/domain.go:73]: It shouldn't have any leaks.

**Pull request merged**